internal/events/subscription_manager.go

```diff
@@ -275,47 +290,93 @@ func (sm *subscriptionManager) parseSubscriptionDef(ctx context.Context, subDef
 		}
 	}
 
-	var tagFilter *regexp.Regexp
-	if filter.Tag != "" {
-		tagFilter, err = regexp.Compile(filter.Tag)
-		if err != nil {
-			return nil, i18n.WrapError(ctx, err, i18n.MsgRegexpCompileFailed, "filter.tag", filter.Tag)
-		}
+	sub = &subscription{
+		dispatcherElection: make(chan bool, 1),
+		definition:         subDef,
+		eventMatcher:       eventFilter,
 	}
 
-	var groupFilter *regexp.Regexp
-	if filter.Group != "" {
-		groupFilter, err = regexp.Compile(filter.Group)
-		if err != nil {
-			return nil, i18n.WrapError(ctx, err, i18n.MsgRegexpCompileFailed, "filter.group", filter.Group)
+	if (filter.Message != fftypes.MessageFilter{}) {
+		var tagFilter *regexp.Regexp
+		if filter.Message.Tag != "" {
+			tagFilter, err = regexp.Compile(filter.Message.Tag)
+			if err != nil {
+				return nil, i18n.WrapError(ctx, err, i18n.MsgRegexpCompileFailed, "filter.Message.tag", filter.Message.Tag)
+			}
 		}
-	}
 
-	var topicsFilter *regexp.Regexp
-	if filter.Topics != "" {
-		topicsFilter, err = regexp.Compile(filter.Topics)
-		if err != nil {
-			return nil, i18n.WrapError(ctx, err, i18n.MsgRegexpCompileFailed, "filter.topics", filter.Topics)
+		var groupFilter *regexp.Regexp
+		if filter.Message.Group != "" {
+			groupFilter, err = regexp.Compile(filter.Message.Group)
```

Comment:
I think we need to handle `filter.DeprecatedGroup` to refer to the old JSON input fields.

Either by referencing both here (with a deprecation warning in the logs on the latter), or at a point up the stack (that's common to durable subscriptions and ephemeral) we copy the value over before we enter this code.